core/src/main/java/org/sonarsource/sonarlint/core/telemetry/TelemetryLocalStorage.java

```diff
@@ -43,12 +45,42 @@
   private int showHotspotRequestsCount;
   private int taintVulnerabilitiesInvestigatedLocallyCount;
   private int taintVulnerabilitiesInvestigatedRemotelyCount;
+  private final Collection<String> explicitlyEnabledRules;
```

Comment:
IMO activated/deactivated rules should not be accumulated, but only requested when the ping is sent.